airflow/api_fastapi/core_api/serializers/tasks.py

```diff
@@ -0,0 +1,159 @@
+    weight_rule: str | None
+    ui_color: str | None
+    ui_fgcolor: str | None
+    template_fields: list[str] | None
+    downstream_task_ids: list[str] | None
+    doc_md: str | None
+    operator_name: str | None
+    params: abc.MutableMapping | None
+    class_ref: dict | None
+    is_mapped: bool | None
+
+    @field_validator("weight_rule", mode="before")
+    @classmethod
+    def validate_weight_rule(cls, wr: str | PriorityWeightStrategy | None) -> str | None:
+        """Validate the weight_rule property."""
+        if wr is None:
+            return None
+        if isinstance(wr, str):
+            return wr
+        return encode_priority_weight_strategy(wr)
+
+    @field_validator("execution_timeout", mode="before")
+    @classmethod
+    def validate_execution_timeout(cls, et: timedelta | None) -> TimeDeltaResponse | None:
+        """Validate the execution_timeout property."""
+        if et is None:
+            return None
+        return TimeDeltaResponse(
+            days=et.days,
+            seconds=et.seconds,
+            microseconds=et.microseconds,
+        )
+
+    @field_validator("retry_delay", mode="before")
+    @classmethod
+    def validate_retry_delay(cls, rd: timedelta | None) -> TimeDeltaResponse | None:
+        """Validate the retry_delay property."""
+        if rd is None:
+            return None
+        return TimeDeltaResponse(
+            days=rd.days,
+            seconds=rd.seconds,
+            microseconds=rd.microseconds,
+        )
+
+    @field_validator("params", mode="before")
+    @classmethod
+    def get_params(cls, params: abc.MutableMapping | None) -> dict | None:
+        """Convert params attribute to dict representation."""
+        if params is None:
+            return None
+        return {param_name: param_val.dump() for param_name, param_val in params.items()}
+
+    @staticmethod
+    def __get_class_ref(obj) -> dict[str, str | None]:
+        """Return the class_ref dict for obj."""
+        is_mapped_or_serialized = isinstance(obj, (MappedOperator, SerializedBaseOperator))
+
+        module_path = None
+        if is_mapped_or_serialized:
+            module_path = obj._task_module
+        else:
+            module_type = inspect.getmodule(obj)
+            module_path = module_type.__name__ if module_type else None
+
+        class_name = None
+        if is_mapped_or_serialized:
+            class_name = obj._task_type
+        elif obj.__class__ is type:
+            class_name = obj.__name__
+        else:
+            class_name = type(obj).__name__
+
+        return {
+            "module_path": module_path,
+            "class_name": class_name,
+        }
+
+    @staticmethod
+    def assign_type_info(task):
+        """Assign type info properties to task and return it."""
+        if task is None:
+            return None
+        task.class_ref = TaskResponse.__get_class_ref(task)
+        task.is_mapped = isinstance(task, MappedOperator)
+        return task
```

Comment:
You can re-use the `_call_function` validator (put it in a common code place and update signature with generic types (typevars)). This way you can do.
`Annotated[dict, BeforeValidator(_call_function)]`

Anyway we shouldn't have a `staticmethod` to call to populate the field. Lets use validators as we do for other fields. (field or model validator)

Reply:
I've simplified this code, please check it out. Let me know if any changes required, thanks!